### ISSUE TITLE
Teleport commands fail on CentOS 7

### ISSUE THREAD

**Issue by SMillerDev** [gravitational/teleport#9028]
### Description

**What happened**:
After updating to Teleport 8.0 on CentOS 7 Teleport fails to start.
```
user@host:~# teleport -h
teleport: /lib64/libc.so.6: version `GLIBC_2.18' not found (required by teleport)
teleport: /lib64/libc.so.6: version `GLIBC_2.25' not found (required by teleport)
```

**What you expected to happen**:
Teleport to start as it did with 7.x.

### Reproduction Steps
As minimally and precisely as possible, describe step-by-step how to reproduce the problem.
1. Update teleport install on CentOS 7 to Teleport 8.0
2.
3.

### Server Details
- Teleport version (run `teleport version`): 8.0.0-1
- Server OS (e.g. from `/etc/os-release`): CentOS 7
- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware): AWS
- Additional details: Using the teleport Yum repo

### Client Details
- Tsh version (`tsh version`): Teleport v7.3.3 git: go1.17.2
- Computer OS (e.g. Linux, macOS, Windows): macOS
- Browser version (for UI-related issues): Firefox 94.0.1 (64-bit)
- Installed via (e.g. apt, yum, brew, website download): brew
- Additional details: N/A

### Debug Logs
Please include or attach debug logs, when appropriate. Obfuscate sensitive information!
- Start Teleport with --debug flag (`teleport --debug`)
- Run tsh with --debug flag (`tsh --debug`)

**Comment by bcg62** [comment]
I am also seeing this

**Comment by savv3** [comment]
Same here...

**Comment by Roguyt** [comment]
Another issue that is kinda linked to this one is the fact that the deb repository only offer the 8.0 version as of today.
Meaning that:
1. Upgrade are breaking if you don't go version by version using deb from Teleport website
2. We can't rollback this version because of the encountered error.

**Comment by webvictim** [comment]
In the meantime, the CentOS 6 binaries should work: https://get.gravitational.com/teleport-v8.0.0-linux-amd64-centos6-bin.tar.gz

**Comment by WilliamLoy** [comment]
Customers who have opted to use rpm packages over binary installs are having to spend time creating their own rpms . For consistency with previous releases it would be good to include rpms to accompany any tar releases.

**Comment by Roguyt** [comment]
The issue isn't really fixed yet.
Yes there is a separate build for CentOS, but if you are using the deb repository how can we opt to use the CentOS 7 version ?

**Comment by SMillerDev** [comment]
Wait, how would you get centos binaries to even install on a Debian based system?

**Comment by Roguyt** [comment]
@SMillerDev You cannot. But this issue isn't linked directly to CentOS.

You can also replicate this issue on some Debian version which are bundled with an old version of GLIBC.
My point is at any point the build version of teleport may break for obscure reason.

**Comment by zmb3** [comment]
Correct, the only thing specific about CentOS with this issue is that CentOS ships with an old version of GLIBC.

Perhaps a more appropriate name for the CentOS binaries would be "Teleport - old GLIBC" or something. In any case, users who wish to run Teleport on systems with such an old version of GLIBC will be now be able to use these binaries when they ship.

**Comment by Roguyt** [comment]
> Correct, the only thing specific about CentOS with this issue is that CentOS ships with an old version of GLIBC.
> 
> Perhaps a more appropriate name for the CentOS binaries would be "Teleport - old GLIBC" or something. In any case, users who wish to run Teleport on systems with such an old version of GLIBC will be now be able to use these binaries when they ship.

Would it be possible to add a libc version requirement to the .deb so that system which aren't compatible won't update using the deb repo ?

**Comment by zmb3** [comment]
> Would it be possible to add a libc version requirement to the .deb so that system which aren't compatible won't update using the deb repo?

That seems like a reasonable request to me. Would you mind creating a separate issue so this discussion doesn't get lost in an old closed issue?

**Comment by SMillerDev** [comment]
@zmb3 when will these fixed releases show up in the RPM repo?

**Comment by webvictim** [comment]
AFAIK there are currently no plans to create separate CentOS 6/7 RPMs - the binary tarballs for those distributions should be used instead.

There is a separate issue tracking adding a glibc dependency to avoid installing the current DEB/RPM files on systems which cannot support them: https://github.com/gravitational/teleport/issues/9247

**Comment by SMillerDev** [comment]
That should probably be noted in the install instructions because https://goteleport.com/docs/installation/ seems to suggest to use the yum repo (which works as long as you don't install 8.x releases)

**Comment by savv3** [comment]
Agree with @SMillerDev. Another suggestion would be to have different yum repos for different major versions, so that one could upgrade major version, once you had tested on a test/dev machine. 
MariaDB, for example, uses this approach with their yum repo, and it works very well.
I just have to enable/disable repos in yum config to upgrade and do yum update, to install the next major version.
